### PR TITLE
Add PR template

### PR DESCRIPTION
**Pull request by aiyengar2**
Related Issue: https://github.com/rancher/rancher/issues/29446

(cherry picked from commit 948de0b3c56ce794ebf6d541a223fd1cd6ca8a93)

Original PR (merged to the wrong branch): https://github.com/rancher/charts/pull/774

### CODE REVIEW

**Comment by aiyengar2** [comment]
Missed this minor comment on the [Github docs](https://docs.github.com/en/free-pro-team@latest/github/building-a-strong-community/creating-a-pull-request-template-for-your-repository):

```
Templates are available to collaborators when they are merged into the repository's default branch.
```

Merging to dev-v2.5-source didn't do anything 😓

**Review by cbron — COMMENTED** [review]
Didn't realize we can't do branch specific pull request templates. Thats unfortunate, because master is still used by our partners and adding this there would be quite confusing.

File: .github/workflows/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
+#### Pull Request Checklist ####
+
```
Comment (aiyengar2) [inline .github/workflows/pull_request_template.md:2]:
```suggestion
**Note: This checklist is only applicable to charts that are being added to the new `Apps & Marketplace` UI.**
```

File: .github/workflows/pull_request_template.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
+#### Pull Request Checklist ####
+**Note: This checklist is only applicable to charts that are being added to the new `Apps & Marketplace` UI.**
```
Comment (aiyengar2) [inline .github/workflows/pull_request_template.md:2]:
```suggestion

**Note: This checklist is only applicable to charts that are being added to the new `Apps & Marketplace` UI.**

```

**Review by cbron — COMMENTED** [review]
@PennyScissors would you agree that this will be confusing for partners to see when submitting a PR to master ? I think once everything is migrated to partner-charts we can do it.

**Comment by aiyengar2** [comment]
@cbron I agree that it would be confusing. At this point, since we're already starting work on the charts script rewrite, I think it would be better to take the second option in the original issue and close out this PR instead:
> - Modify Github PR workflow: fail if the tgz file generated by make charts for packages that were modified in this pull request already exists in the main branch.

We could just execute that as part of https://github.com/rancher/rancher/issues/28730 instead.

**Comment by cbron** [comment]
Yes that sounds like the correct behavior, failing if same version is in the release branch.

**Pull request closed without merging**